src/Http/RequestFactory.php

```diff
@@ -82,6 +82,7 @@ public function createHttpRequest()
 
 		// path & query
 		$requestUrl = isset($_SERVER['REQUEST_URI']) ? $_SERVER['REQUEST_URI'] : '/';
+		$requestUrl = preg_replace('#^\w++://[^/]++#', '', $requestUrl);
```

Comment:
Is this actually RFC compliant? What if I send mismatching Host header or use different protocol than used in here? Shouldn't we strip it only if it is a match and throw exception otherwise?